Improve get ticket for the authentication advice

smasue opened the pull request:
I removed entirely the getTicket method to replace it with the one from the AuthenticationService. However this can lead to incompatibilities with the old versions of Alfresco. I didn't know what you want to maintain or not.

Also I have big doubts of what the getCurrentTicket will return if the person is not authenticated.

dgradecak commented:
Thank you for the contribution.

I think all is good and that you should not worry about that scenario. The worst case is if someone create the webscript descriptor with authentication none and actually create an @AlfrescoAuthentication with anything else than none, in that case the "getTicket" will fail simply because there is no authentication provided, and that is fine. If the developer is doing that, IMHO he will discover very quickly that without authentication it cannot work. Even if the ticket is provided in the URL. The old way, if someone provides a ticket in the URL it would authenticate "transparently".

So, in order to answer your doubt, actually it will fail to get the ticket only if the webscript is set to auth "none" and the AlfrescoAuthentication annotation is set to anything else than NONE, and honestly I think that is as it should be.

Also, remeber that guest authentication is different than none authentication. That is why, having the authentication user on the mvc webscript level is good and if for any case a none is needed, than no @AlfrescoAuthentication should be used without having any kind of authentication executed previously (like an interceptor or so).

smasue commented:
I see.
In that case do we even need to get and then check the ticket? I'm wondering. 
In another version locally, I went further by removing the ticket thingy. I removed the last "else" condition and added a condition that throws an exception if the guest was not allowed. 

`AuthenticationType.GUEST.equals(authenticationType) && !authenticationService.guestUserAuthenticationAllowed()`

dgradecak commented:
can you show me what exactly you did?

smasue commented:
Something like this:
https://gist.github.com/smasue/3bf2d762853664073e27
